src/System.Collections.NonGeneric/tests/Hashtable/SynchronizedTests.cs

```diff
@@ -0,0 +1,251 @@
+            }
+
+            //now we test the modifying methods
+            _hsh2.Remove("Key_1");
+            Assert.False(_hsh2.ContainsKey("Key_1"));
+            Assert.False(_hsh2.ContainsValue("Value_1"));
+
+            _hsh2.Add("Key_1", "Value_1");
+            Assert.True(_hsh2.ContainsKey("Key_1"));
+            Assert.True(_hsh2.ContainsValue("Value_1"));
+
+            _hsh2["Key_1"] = "Value_Modified_1";
+            Assert.True(_hsh2.ContainsKey("Key_1"));
+            Assert.False(_hsh2.ContainsValue("Value_1"));
+            ///////////////////////////
+
+            Assert.True(_hsh2.ContainsValue("Value_Modified_1"));
+            hsh3 = Hashtable.Synchronized(_hsh2);
+
+            //we are not going through all of the above again:) we will just make sure that this syncrnized and that 
+            //values are there
+            Assert.Equal(hsh3.Count, hsh1.Count);
+            Assert.True(hsh3.IsSynchronized);
+
+            _hsh2.Clear();
+            Assert.Equal(_hsh2.Count, 0);
+
+            //[] Synchronized returns a HT that is thread safe
+            // We will try to test this by getting a number of threads to write some items
+            // to a synchronized IList
+            hsh1 = new Hashtable();
+            _hsh2 = Hashtable.Synchronized(hsh1);
+
+            workers = new Task[iNumberOfWorkers];
+            for (int i = 0; i < workers.Length; i++)
+            {
+                var name = "Thread worker " + i;
+                ts1 = new Action(() => AddElements(name));
+                workers[i] = Task.Run(ts1);
+            }
+
+            Task.WaitAll(workers);
+
+            //checking time
+            Assert.Equal(_hsh2.Count, _iNumberOfElements * iNumberOfWorkers);
+
+            for (int i = 0; i < iNumberOfWorkers; i++)
+            {
+                for (int j = 0; j < _iNumberOfElements; j++)
+                {
+                    strValue = "Thread worker " + i + "_" + j;
+                    Assert.True(_hsh2.Contains(strValue));
+                }
+            }
+
+            //I dont think that we can make an assumption on the order of these items though
+            //now we are going to remove all of these
+            workers = new Task[iNumberOfWorkers];
+            for (int i = 0; i < workers.Length; i++)
+            {
+                var name = "Thread worker " + i;
+                ts1 = new Action(() => RemoveElements(name));
+                workers[i] = Task.Run(ts1);
+            }
+
+            Task.WaitAll(workers);
+
+            Assert.Equal(_hsh2.Count, 0);
+            Assert.False(hsh1.IsSynchronized);
+            Assert.True(_hsh2.IsSynchronized);
+
+            //[] Tyr calling Synchronized with null
+            Assert.Throws<ArgumentNullException>(() =>
+                             {
+                                 Hashtable.Synchronized(null);
+                             }
+            );
+        }
+
+
+        void AddElements(string strName)
```

Comment:
The style guidelines call for visibility modifiers always being specified (also applies to the next method).  And please remove the extra blank line above this method.